### PR TITLE
'Instructor only' visibility is buggy #6078

### LINKED ISSUE #6078: 'Instructor only' visibility is buggy
image

### PR DESCRIPTION
**Pull request by YongJieYongJie**
Fixes #6078

### CODE REVIEW

**Comment by damithc** [comment]
How's this going?

**Comment by YongJieYongJie** [comment]
@damithc I'm reviewing the GodMode changes, will be ready after that

**Comment by YongJieYongJie** [comment]
@damithc @wkurniawan07 Ready for review!

**Comment by damithc** [comment]
@wkurniawan07 let's push this out soon. I promised to fix this by end of this week end.

File: src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js [hunk 1]
```diff
@@ -196,6 +196,28 @@ function updateVisibilityCheckboxesDiv($containingForm) {
...
     disableRowsForSpecificGiverRecipientCombinations($containingForm);
+
+    // handles edge case for Team Contribution Question:
+    // normal behavior is that all hidden checkboxes are unchecked, but Team Contribution Question expect even the hidden
+    // Recipient's Team Members can see answer checkbox to be checked
+    fixCheckboxValuesForTeamContribQuestion($containingForm);
+}
+
+/**
+ * Ensure the hidden checkbox for Recipient's Team Members can see answer is consistent with Recipient can see answer
```
Comment (damithc) [inline src/main/webapp/js/instructorFeedbackEdit/visibilityOptions.js]: Ensures

File: src/main/webapp/js/instructorFeedbackEdit.js [hunk 2]
```diff
@@ -319,7 +319,8 @@ function enableQuestion(questionNum) {
     $('#' + FEEDBACK_QUESTION_EDITTYPE + '-' + questionNum).val('edit');
     $('#button_question_submit-' + questionNum).show();
 
-    showVisibilityCheckboxesIfCustomOptionSelected($currentQuestionTable);
+    $currentQuestionForm = $currentQuestionTable.closest('form');
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorFeedbackEdit.js]: Is this variable defined before? May need to add `var $currentQuestionForm = ...` if not.

**Comment by wkurniawan07** [comment]
Confirmed that the added test failed before the fix, passes after, and also that the visibility bug as shown in the issue thread is fixed.
@YongJieYongJie other than the minor comments, you need to merge this to `release` branch instead of `master` branch.

**Comment by damithc** [comment]
Let's finish this soon. I promised to fix this bug over the weekend. :-p

**Comment by damithc** [comment]
Ready?

**Comment by YongJieYongJie** [comment]
@damithc Yup, ready!

**Comment by damithc** [comment]
@wkurniawan07 let's release this quickly.

**Comment by wkurniawan07** [comment]
@damithc released as V5.89.02.

**Comment by damithc** [comment]
Deployed. Thanks both.

**Pull request merged**